components/server/ee/src/prebuilds/github-app.ts

```diff
@@ -33,46 +33,54 @@ import { Options, ApplicationFunctionOptions } from 'probot/lib/types';
  * look at those values to begin with.
  */
 
-decorate(injectable(), Probot)
-
 @injectable()
-export class GithubApp extends Probot {
+export class GithubApp {
     @inject(AppInstallationDB) protected readonly appInstallationDB: AppInstallationDB;
     @inject(UserDB) protected readonly userDB: UserDB;
     @inject(TracedWorkspaceDB) protected readonly workspaceDB: DBWithTracing<WorkspaceDB>;
     @inject(MessageBusIntegration) protected readonly messageBus: MessageBusIntegration;
     @inject(GithubAppRules) protected readonly appRules: GithubAppRules;
     @inject(PrebuildManager) protected readonly prebuildManager: PrebuildManager;
 
+    readonly server: Server | undefined;
+
     constructor(
         @inject(Env) protected readonly env: Env,
         @inject(PrebuildStatusMaintainer) protected readonly statusMaintainer: PrebuildStatusMaintainer,
     ) {
-        super({
-            id: env.githubAppAppID,
-            privateKey: GithubApp.loadPrivateKey(env.githubAppCertPath),
-            secret: env.githubAppWebhookSecret,
-            logLevel: env.githubAppLogLevel as Options["logLevel"]
-        });
-        log.debug("Starting GitHub app integration", {
-            id: env.githubAppAppID,
-            cert: env.githubAppCertPath,
-            secret: env.githubAppWebhookSecret
-        })
-
-        this.load(this.buildApp.bind(this));
+        if (env.githubAppEnabled) {
+            this.server = new Server({
+                Probot: Probot.defaults({
+                    appId: env.githubAppAppID,
+                    privateKey: GithubApp.loadPrivateKey(env.githubAppCertPath),
+                    secret: env.githubAppWebhookSecret,
+                    logLevel: env.githubAppLogLevel as Options["logLevel"]
+                })
+            })
+            log.debug("Starting GitHub app integration", {
+                appId: env.githubAppAppID,
+                cert: env.githubAppCertPath,
+                secret: env.githubAppWebhookSecret
+            })
+            this.server.load(this.buildApp.bind(this));
+        }
     }
 
-    protected async buildApp(options: ApplicationFunctionOptions) {
-        const app = options.app;
+    protected async buildApp(app: Probot, options: ApplicationFunctionOptions) {
         this.statusMaintainer.start(async id => (await app.auth(parseInt(id))) as any as Octokit);
         // this.queueMaintainer.start();
 
         // Backward-compatibility: Redirect old badge URLs (e.g. "/api/apps/github/pbs/github.com/gitpod-io/gitpod/5431d5735c32ab7d5d840a4d1a7d7c688d1f0ce9.svg")
-        options.getRouter('/pbs').get('/*', async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+        options.getRouter && options.getRouter('/pbs').get('/*', async (req: express.Request, res: express.Response, next: express.NextFunction) => {
```

Comment:
When is `getRouter` undefined and why would we want to silently not register the routes in such a case?

Reply:
This is a misleading declaration. `getRouter` is always defined. I'll add a comment on this.
```
    async load(appFn) {
        await appFn(this.probotApp, {
            getRouter: (path) => this.router(path),
        });
    }
```